### ISSUE TITLE
Button group pie charts

### ISSUE THREAD

**Issue by starSweeper** [starSweeper/visibilityMachine#23]
For each button group (Clear-Smoky-Foggy/Good-Bad) add a pie chart that shows what percentage of labeled images are being put into each category. Also include skipped images and remaining images in these pie charts. These will be updated as images are labeled.

**Comment by starSweeper** [comment]
**Update:** I am using the library [JFreeCharts](https://github.com/jfree/jfreechart), and this [Stack Overflow Question](https://stackoverflow.com/questions/26014528/can-someone-please-provide-step-by-step-instructions-for-using-jfreechart-in-an) was incredibly valuable when trying to figure out how to install it properly. I am also following[ this tutorial](http://www.java2s.com/Code/Java/Chart/JFreeChartPieChartDemo1.htm).

**Comment by starSweeper** [comment]
**Update:** You need a [ChartPanel](https://stackoverflow.com/questions/7945565/add-a-jfreechart-in-to-jpanel ) to display a JFreeChart.

**Comment by starSweeper** [comment]
**Update:** The charts have been added, but somehow when I was reorganizing the layout to allow the addition of new components, the image gallery (#12) disappeared! The code is there, the gallery just doesn't appear in the layout. I need to find it before I can declare this issue complete.

I also want to make it so that categories with zero images do not appear on the chart until at least one image belongs to that group.

**Comment by starSweeper** [comment]
**Update:** I solved the mystery of the disappearing image gallery. I was accidentally setting the layout of the containing panel (centerWest) AFTER adding the image gallery to the north part of that panel.

Now, the removing empty category thing should be simple, and this issue will soon be closed.

**Comment by starSweeper** [comment]
**Update:** Okay, I removed the empty categories.
```
 private PieDataset updateCSFDataSet(){
        DefaultPieDataset dataSet = new DefaultPieDataset();
        if(clearCount > 0) {
            dataSet.setValue("Clear", clearCount);
        }
        if(smokyCount > 0) {
            dataSet.setValue("Smoky", smokyCount);
        }
        if(foggyCount > 0) {
            dataSet.setValue("Foggy", foggyCount);
        }
        if(skipCount > 0) {
            dataSet.setValue("Skipped", skipCount);
        }
        if(unlabeledCount > 0) {
            dataSet.setValue("Unlabeled", unlabeledCount);
        }
        return dataSet;
    }

    private PieDataset updateGBDataSet(){
        DefaultPieDataset dataSet = new DefaultPieDataset();
        if(goodCount > 0) {
            dataSet.setValue("Good", goodCount);
        }
        if(badCount > 0) {
            dataSet.setValue("Bad", badCount);
        }
        if(skipCount > 0) {
            dataSet.setValue("Skipped", skipCount);
        }
        if(unlabeledCount > 0) {
            dataSet.setValue("Unlabeled", unlabeledCount);
        }
        return dataSet;
    }
```
The chart is still too wide though. I have about 12 minutes until it's time to go home. Lets see if I can fix the dimensions before then?

**Comment by starSweeper** [comment]
**Update:** It may be useful to note that [some layout managers do not care about the specified size of your components](https://thebadprogrammer.com/swing-layout-manager-sizing/). This seems to be why I am having so much trouble with #39.

**Comment by starSweeper** [comment]
**Update:** Basically, BorderLayout is messing everything up. And I'm closing this issue even though the chart is still too wide because changing the layout now would be a huge undertaking :(

**Comment by starSweeper** [comment]
**Update:** I added category totals to each slice label

**Comment by starSweeper** [comment]
**Update:** A separate issue has been opened to fix pie chart size (#40)

**Comment by starSweeper** [comment]
**Update:** It turns out I don't need any if statements to remove zero values. There is already a function for that.
![image](https://user-images.githubusercontent.com/26500355/37121970-12270404-221c-11e8-851a-af789d96ee7a.png)
![image](https://user-images.githubusercontent.com/26500355/37122034-3c14eda8-221c-11e8-81a1-dec0533ce7b1.png)
